src/app/parcels/ActionBar/Actions.tsx

```diff
@@ -29,32 +30,36 @@ export type ActionName =
     | "Download Driver Overview"
     | "Delete Parcel";
 
-type AvailableActionsType = {
-    [actionKey in ActionName]: {
-        newStatus: StatusType;
-    };
+type ActionTypes = {
+    actionName: ActionName;
+    newStatus: StatusType;
+    availableToVolunteers: boolean;
 };
```

Comment:
if we wanted to have the availability of actions dependant on other roles, we'd have to create more properties in this object for each role. this isn't ideal. maybe we could have a switch case statement elsewhere (perhaps in roles?) that takes in a role and returns which actions are available to that role?

Reply:
what about just having rolesAvailableTo = [...] as a property on this?

Reply:
yeah I think that is also a better solution than currently, but I think it might get a bit messy if we add lots more roles, I could see the possibility of someone forgetting to copy and paste a new role into another action if ygm. If we were to use a switch case we could also have a nice default that allows all roles, which might be quite nice and neat. However, I don't think there's that much difference, and this also has the advantage of not having to make a seperate function.

Reply:
Yeah I defo see what you mean and that's definitely an advantage of doing it that way. Though you still have the issue with forgetting to add the role onto the switch statement & it may be harder to couple the two together (filter by the results of the switch statement). That being said, there's pros and cons with both & you've convinced me that I'd be happy to see either (or a 3rd suggestion)!